package/kernel/mac80211/patches/ath9k/552-ath9k-ahb_of.patch

```diff
@@ -165,17 +165,13 @@
 +	struct ath9k_platform_data *pdata;
 +	const struct of_device_id *match;
 +	const struct of_ath_ahb_data *data;
-+	u8 led_pin;
 +
 +	match = of_match_device(of_ath_ahb_match, &pdev->dev);
 +	data = (const struct of_ath_ahb_data *)match->data;
 +
 +	pdata = dev_get_platdata(&pdev->dev);
 +
-+	if (!of_property_read_u8(pdev->dev.of_node, "qca,led-pin", &led_pin))
-+		pdata->led_pin = led_pin;
-+	else
-+		pdata->led_pin = -1;
++	pdata->led_pin = -1;
```

Comment:
Are you sure this is correct?

I think the `led_pin` option will configure the GPIO controller in the ath9k module your change to the `target/linux/ath79/dts/ar9344_mikrotik_routerboard-951ui-2hnd.dts` uses the GPIO controller of the SoC.

Reply:
This is how it was with the original Mach file.

Reply:
further discussion: https://github.com/openwrt/openwrt/pull/12226#issuecomment-2227601735

I remember the situation being the same with ramips. [mt76ath9k] phys creating their own LED entries while doing nothing because a proper GPIO is not assigned.

Reply:
DO you know if this change will work?

Reply:
I have not tested but I believe so.

@kempniu knows more.

Reply:
1. This change lacks the `linux,default-trigger = "phy0tpt";` line that I [proposed](https://github.com/openwrt/openwrt/pull/12226#issuecomment-2228104892) in the DTS snippet to use in case `qca,led-pin` needs to go away.  That line seems to be commonly used for other Mikrotik devices (see `git grep phy0tpt -- ':**/*mikrotik*.dts'`), so I'm not sure why it was left out from this PR - that breaks the WLAN LED because there are no longer any defaults in effect for the RB951Ui-2HnD that would make it work...  Either the default trigger needs to be set in the DTS file or something else needs to set the trigger for the `green:wlan` LED from userspace.
2. This change "works", as in: assuming the above trigger issue is addressed, the WLAN LED will work on the RB951Ui-2HnD, but the `ath9k-phy0` LED will still be created with the default (wrong) GPIO number, potentially causing confusion.

Reply:
I'll add 1.

As for 2, that's already the case with multiple devices. It's an open question whether or not the ath9k or mt76 driver should be used to control LED GPIO.

Reply:
I suggest keeping this hack. This is not a significant change, perhaps there are still some users who want this feature. BTW, mt76 also can configure the LED pin. Other parts looks good to me.
https://git.openwrt.org/?p=openwrt%2Fopenwrt.git&a=search&h=HEAD&st=grep&s=led-sources

Reply:
It begs the question, should more of led-pin be used instead of normal GPIO?